Sources/EosioSwift/EosioTransaction/EosioTransaction.swift

```diff
@@ -455,23 +460,44 @@ public class EosioTransaction: Codable {
         guard let rpcProvider = rpcProvider else {
             return completion(.failure(EosioError(.eosioTransactionError, reason: "No rpc provider available")))
         }
-
-        let requestParameters = EosioRpcBlockInfoRequest(blockNum: blockNum)
-
-        rpcProvider.getBlockInfoBase(requestParameters: requestParameters, completion: { [weak self] (blockResponse) in
-            guard let strongSelf = self else {
-                return completion(.failure(EosioError(.getBlockError, reason: "self does not exist")))
-            }
-            switch blockResponse {
-            case .failure(let error):
-                completion(.failure(error))
-            case .success(let block):
-                // set tapos fields and return
-                strongSelf.refBlockNum = UInt16(block.blockNum.value & 0xffff)
-                strongSelf.refBlockPrefix = block.refBlockPrefix.value
-                return completion(.success(true))
-            }
-        })
+        
+        // Support for the old node version
+        if chainVersion == .v2 {
```

Comment:
I could also abstract those EosioRpcBlockInfoRequest and EosioRpcBlockRequest with protocol to provide a unified body for the closures, which would prevent duplication but would be a bit harder to remove later on when the support is dropped.

Reply:
It could be something like: 

```swift
protocol BlockInfo {
  var blockNum: Int { get }
  var blockPrefix: String { get }
}
```